src/main/scala/firrtl/util/BackendCompilationUtilities.scala

```diff
@@ -24,17 +37,25 @@ trait BackendCompilationUtilities {
     out.close()
   }
 
-  /** Create a temporary directory with the prefix name. Exists here because it doesn't in Java 6.
+  /** Create a test directory
+    *
+    * Will create outer directory called testName then inner directory based on
+    * the current time
     */
-  def createTempDirectory(prefix: String): File = {
-    val temp = File.createTempFile(prefix, "")
-    if (!temp.delete()) {
-      throw new IOException(s"Unable to delete temp file '$temp'")
-    }
-    if (!temp.mkdir()) {
-      throw new IOException(s"Unable to create temp directory '$temp'")
+  def createTestDirectory(testName: String): File = {
+    val outer = new File(TestDirectory, testName)
+    val testDir = {
+      val dir: Future[File] = Future {
+        var _dir = new File(outer, timeStamp)
+        while (!_dir.mkdirs()) {
+          Thread.sleep(1)
+          _dir = new File(outer, timeStamp)
+        }
+        _dir
+      }
+      Await.result(dir, 10 seconds)
     }
```

Comment:
I think we should use standard utility methods where they are available. You can use time as part of the prefix if you like, but be careful about making the absolute path too long.

I can see an argument that this isn't really a temporary directory - it's not clear when it can be deleted - but we don't really care what the name is, as long is it's unique.